Cython/Utility/ModuleSetupCode.c

```diff
@@ -162,6 +162,9 @@
   #define CYTHON_USE_DICT_VERSIONS 0
   #undef CYTHON_USE_EXC_INFO_STACK
   #define CYTHON_USE_EXC_INFO_STACK 0
+  #if PYPY_VERSION_HEX < 0x07030900
+    #define CYTHON_AVOID_SLOT_DOC
+  #endif
```

Comment:
I implemented the needed pieces to properly use `PyWrapperDescrObject.d_base.doc` for the docstring in PyPy 7.3.9.

Reply:
The usual pattern is to set the define to 0 or 1 and test if it's defined (or explicitly undefine it), so that users can override it by setting it externally via CFLAGS, and different Python implementations can define it to their needs.

Also, the name suggests to me that this removes all usages of the `tp_doc` slot, which it doesn't.
`CYTHON_UPDATE_DESCRIPTOR_DOC` ?

Reply:
updating to use the positive `CYTHON_UPDATE_DESCRIPTOR_DOC` rather than the negative `CYTHON_AVOID_SLOT_DOC`